core/src/main/scala/cats/Monad.scala

```diff
@@ -40,6 +40,18 @@ trait Monad[F[_]] extends FlatMap[F] with Applicative[F] {
    * Collects the results into an arbitrary `Alternative` value, such as a `Vector`.
    * This implementation uses append on each evaluation result,
    * so avoid data structures with non-constant append performance, e.g. `List`.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.{Id, Monad}
+   * scala> import cats.data.StateT
+   * scala> import cats.syntax.all._
+   * scala> val appendStr: StateT[Id, String, Unit] = StateT.modify(_ + "a")
+   * scala> val appendStrAndGet: StateT[Id, String, String] = appendStr *> StateT.get
+   * scala> val (result, agg) = appendStrAndGet.whileM[Vector](StateT.inspect(i => !(i.length >= 5))).run("")
+   * result: String = aaaaa
+   * agg: Vector[String] = Vector(a, aa, aaa, aaaa, aaaaa)
```

Comment:
Instead of `StateT[Id, ...]` can't we just use `State`? The issue with `StateT[Id, ...]` is that actually it is not stacksafe.

Reply:
Sure, should I use `State` also in all other examples that uses `StateT` or just this one?

Reply:
Yes. You should never use `StateT[Id`.

Reply:
Done. I'm using `State` now, but there's some results that's previous result + 1, like in `iterateUntilM`. Is that okay?